samza-core/src/main/java/org/apache/samza/processor/StreamProcessor.java

```diff
@@ -178,20 +248,25 @@ public void start() {
    */
   public void stop() {
     synchronized (lock) {
-      boolean containerShutdownInvoked = false;
-      if (container != null) {
+      List<State> expectedStates = ImmutableList.of(State.NEW, State.RUNNING, State.STARTED, State.IN_REBALANCE);
```

Comment:
Stop can be called in any state. So this check just needs to see if you are already STOPPING.